optimizer-hints.md

```diff
@@ -24,50 +24,65 @@ For example, the following query uses three different hints:
 select /*+ USE_INDEX(t1, idx1), HASH_AGG(), HASH_JOIN(t1) */ count(*) from t t1, t t2 where t1.a = t2.b;
 ```
 
-Currently, TiDB supports two categories of hints, which are slightly different in usage. The first category of hints is used to control the optimizer's behaviors, such as [`/*+ HASH_AGG() */`](#hash_agg); the second category of hints is used to set operation parameters for a single query, such as [`/*+ MEMORY_QUOTA(1024 MB)*/`](#memory_quotan).
+Optimizer hints can be used along with `Explain`/`Explain Analyze`. These two commands can check whether optimizer hints have taken effect on the queries as expected. If optimizer hints contains syntax errors or are not applicable to the statements, the query is executed without the optimizer hints. It does not report an error on the optimizer hints, but only record a warning. You can view the detailed information after the query is completed by running the `Show Warnings` command.
```

Comment:
The first sentence implies somewhat that the purpose of the hint is tied to EXPLAIN / EXPLAIN ANALYZE, when it is not really - the purpose is to change the actual query, EXPLAIN is an observation tool (explained in second sentence). I think someone using hints should understand what EXPLAIN does somewhat, so I've reworded it a little and linked to explain if they don't.

```suggestion
How optimizer hints affect query execution plans can be observed in the output of [`EXPLAIN`](/sql-statements/sql-statement-explain.md) and [`EXPLAIN ANALYZE`](/sql-statements/sql-statement-explain-analyze.md).

An incorrect or incomplete hint will not result in a statement error. This is because hints are intended to have only a _hint_ (suggestion) semantic to query execution. Similarly, TiDB will at most return a warning if a hint is not applicable.
```

Reply:
@eurekaka PTAL the comment above.

Reply:
LGTM